### PR TITLE
[9.0][REF]web_ir_actions_act_window_message: fix refresh view

### PR DESCRIPTION
**Pull request by zakiuu**
Take into consideration the view type when refreshing it,
the `form` view uses `recursive_reload`, and list view uses
`reload_content`

### CODE REVIEW

**Comment by zakiuu** [comment]
closes https://github.com/OCA/web/issues/976

File: web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js [hunk 1]
```diff
@@ -25,8 +25,19 @@ odoo.define('web.web_ir_actions_act_window_message', function(require)
                     text: action.close_button_title || _t('Close'),
                     click: function() {
                         // refresh the view before closing the dialog
-                        self.inner_widget.active_view
-                            .controller.recursive_reload();
+                        var controller = self.inner_widget.active_view.controller;
```
Comment (hbrunn) [inline web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js]: that's what I meant with refactoring: Move this to its own function, something like `ir_actions_act_window_message_refresh` (to be sure we won't overwrite some action manager function) and do this there.

File: web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js [hunk 2]
```diff
@@ -25,8 +25,19 @@ odoo.define('web.web_ir_actions_act_window_message', function(require)
...
+                        // recursive_reload function is available
+                        // in the form view
+                        if (controller.recursive_reload)
+                        {
+                            controller.recursive_reload();
+                        }
+                        // reload_content function is available
+                        // in the list view
+                        if (controller.reload_content)
```
Comment (hbrunn) [inline web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js]: pull this up to have an `else if` branch above

File: web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js [hunk 3]
```diff
@@ -25,8 +25,19 @@ odoo.define('web.web_ir_actions_act_window_message', function(require)
...
+                        {
+                            controller.reload_content()
+                        }
```
Comment (hbrunn) [inline web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js]: and in an `else` branch, you can `console.log` that a refresh was requested, that's nice for the next person using this in a different sort of view

**Review by hbrunn — CHANGES_REQUESTED** [review]

**Comment by zakiuu** [comment]
@hbrunn fixed!

**Review by hbrunn — APPROVED** [review]

**Review by andreparames — APPROVED** [review]

**Comment by zakiuu** [comment]
@hbrunn thanks a lot for your help 👍

**Review by ThomasBinsfeld — APPROVED** [review]

**Comment by zakiuu** [comment]
@jbeficent  Travis seems to be red because of a `Bokeh` library, is there a solution that you can provide for that.

**Comment by andreparames** [comment]
cc @simahawk 

Seems to be the same problem as in https://github.com/OCA/web/pull/852, maybe the same fix is required in v9?

**Comment by andreparames** [comment]
Ah, I think the fix is this: https://github.com/OCA/web/pull/909

@zakiuu maybe you could try cherry-picking that commit here and see if it fixes Travis?

**Comment by zakiuu** [comment]
@andreparames thx for solution, the `Bokeh` issue is gone
@sbidoul still some lint errors but not related to this module, so I guess we can merge

**Review by ThomasBinsfeld — APPROVED** [review]

**Review by benwillig — APPROVED** [review]

**Pull request closed without merging**